### PR TITLE
Ignore try import failure on Windows

### PR DESCRIPTION
**Pull request by andrewkatson**
Fixes https://github.com/bazelbuild/bazel/issues/13443

### CODE REVIEW

File: src/main/cpp/util/file_windows.cc [hunk 1]
```diff
@@ -259,7 +259,15 @@ bool ReadFile(const string& filename, string* content, int max_size) {
...
     return true;
   }
-  return ReadFile(Path(filename), content, max_size);
+
+  // In order for try-imports to be ignored gracefully we need to check for an error with the path and return false
+  // rather than die.
+  std::string errorText;
+  Path path = Path(filename, &errorText);
+  if (!errorText.empty()) {
+      return false;
```
Comment (meteorcloudy) [inline src/main/cpp/util/file_windows.cc:268]: Maybe we can print the error as a warning here?
See https://cs.opensource.google/bazel/bazel/+/master:src/main/cpp/util/logging.h;l=101;bpv=1;bpt=1

File: src/main/cpp/util/path_windows.cc [hunk 2]
```diff
@@ -468,19 +468,27 @@ static char GetCurrentDrive() {
...
+        std::string error;
+        if (!AsAbsoluteWindowsPath(path, &path_, &error)) {
+            if (errorText == nullptr) {
+                BAZEL_DIE(blaze_exit_code::LOCAL_ENVIRONMENTAL_ERROR)
+                        << "Path::Path(" << path
+                        << "): AsAbsoluteWindowsPath failed: " << error;
+            } else {
+                *errorText = error;
+            }
+
```
Comment (meteorcloudy) [inline src/main/cpp/util/path_windows.cc:489]: nit: empty line

Also the indentation is somehow changed and no longer consistent with rest of the code.

File: src/test/cpp/rc_file_test.cc [hunk 3]
```diff
@@ -857,6 +857,18 @@ TEST_F(BlazercImportTest, BazelRcTryImportDoesNotFailForUnreadableFile) {
   ParseOptionsAndCheckOutput(args, blaze_exit_code::SUCCESS, "", "");
 }
 
+#if defined(_WIN32)
```
Comment (meteorcloudy) [inline src/test/cpp/rc_file_test.cc:860]: Nit: Please fix the indentation

**Review by meteorcloudy — COMMENTED** [review]
Nice, thanks!

File: src/main/cpp/util/file_windows.cc (hunk 1 above)
Comment (andrewkatson) [reply src/main/cpp/util/file_windows.cc:269]: Added a warning

File: src/main/cpp/util/path_windows.cc (hunk 2 above)
Comment (andrewkatson) [reply src/main/cpp/util/path_windows.cc]: Fixed. It was a problem with my IDE

File: src/test/cpp/rc_file_test.cc (hunk 3 above)
Comment (andrewkatson) [reply src/test/cpp/rc_file_test.cc:860]: Fixed

**Review by meteorcloudy — APPROVED** [review]
Thanks!

**Pull request closed without merging**